src/Faithlife.Tracing.Http/HttpTracingUtility.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+
+namespace Faithlife.Tracing.Http
+{
+	public static class HttpTracingUtility
+	{
+		public static Func<HttpWebRequest, IDisposable> CreateStartTrace(ITraceProvider traceProvider) => r => SetHttpHeadersAndCreateTrace(traceProvider?.CurrentTrace, r.RequestUri, (k, v) => r.Headers[k] = v);
```

Comment:
This is confusing, especially if you don't know how it will be used. What about an extension method on `ITraceProvider`?

```

Reply:
I don't think I knew that C# supported partial application like this.